protected.go

```diff
@@ -54,28 +69,71 @@ func New(protect Protect, dnsServerIP string) *Protector {
 // Resolve resolves the given address using a DNS lookup on a UDP socket
 // protected by the given Protect function.
 func (p *Protector) Resolve(network string, addr string) (*net.TCPAddr, error) {
-	op := ops.Begin("protected-resolve").Set("addr", addr)
```

Comment:
By moving this down into `resolve`, we lose the ability to mark this op as failed when we get the unsupported network error.